cmd/buildah/rmi.go

```diff
@@ -248,19 +259,21 @@ func removeImage(image *storage.Image, store storage.Store) (string, error) {
 	if _, err := store.DeleteImage(image.ID, true); err != nil {
 		return "", errors.Wrapf(err, "could not remove image %q", image.ID)
 	}
-	for parent != nil {
+	if parent != nil {
 		nextParent, err := getParent(store, parent.TopLayer)
 		if err != nil {
-			return "", err
+			logrus.Debugf("unable to get parent from image %q: %v", image.ID, err)
+			return image.ID, nil
 		}
 		children, err := getChildren(store, parent.TopLayer)
 		if err != nil {
-			return "", err
+			logrus.Debugf("unable to get children from image %q: %v", image.ID, err)
+			return image.ID, nil
 		}
 		// Do not remove if image is a base image and is not untagged, or if
 		// the image has more children.
 		if len(parent.Names) > 0 || len(children) > 0 {
-			return "", nil
+			return image.ID, nil
```

Comment:
We only want to return the image ID if we are actually deleting the image, in this case we are not removing the image, so not returning the image ID is ideal.

Reply:
```
➜  buildah git:(rmi-fix) ✗ sudo buildah images
IMAGE ID             IMAGE NAME                                               CREATED AT             SIZE
e1ddd7948a1c         docker.io/library/busybox:latest                         Aug 1, 2018 06:20      1.38 MB
749ee404bb0b         <none>                                                   Aug 10, 2018 15:41     1.38 MB
➜  buildah git:(rmi-fix) ✗ sudo buildah rmi 749

➜  buildah git:(rmi-fix) ✗ sudo buildah images 
IMAGE ID             IMAGE NAME                                               CREATED AT             SIZE
e1ddd7948a1c         docker.io/library/busybox:latest                         Aug 1, 2018 06:20      1.38 MB
```

In the above instance, the relevant iamge has been deleted, but the relevant imageID is not shown, because the "" is returned here instead of imageID.


```	
if _, err := store.DeleteImage(image.ID, true); err != nil {
		return "", errors.Wrapf(err, "could not remove image %q", image.ID)
	}
```
Image has been removed through the above code.

In this loop, only some operations are performed on the intermediate image. If the question appears, it should not affect the delete command, because the associated image has been deleted.

Reply:
Ah right. Can you change it so we still return the error instead of returning nil. I feel it is best to fail the removal of the intermediate images when one of them throws an error. This way the user knows we tried to delete the cached images but one failed so we stopped trying to remove the other related ones.
So return the id and the error with your wrapped messages.

Reply:
Maybe I don't have enough tea in my system yet, but I'm still seeing nil being returned and not the error.  Did you have a git hiccup?

Reply:
@TomSweeneyRedHat this one is fine, it should be nil. Nothing failed, we just realised that the image is a base image or has more than one child, so we don't want to remove it.